### ISSUE TITLE
Question on sequence diagram with multiple return statements

### ISSUE THREAD

**Issue by limli** [nus-cs2103-AY1819S2/forum#94]
![image](https://user-images.githubusercontent.com/39252823/56656026-fd418c80-66c6-11e9-9c1b-50fee13ad167.png)

This is from sem 1's pyp. 
My question is, how do you represent a return statement inside of a if statement using sequence diagram? I believe "alt"s are used for if statements, but the return makes it a bit awkward. 

![image](https://user-images.githubusercontent.com/39252823/56657088-bd2fd900-66c9-11e9-9f3f-45f6ef207d2a.png)

**Comment by daviddl9** [comment]
My guess would be that it looks something likes this: 
![image](https://user-images.githubusercontent.com/39140938/56657741-688d5d80-66cb-11e9-8aa7-aa58235d5703.png)

**Comment by limli** [comment]
I see. Thank you :)

**Comment by daviddl9** [comment]
@limli I'm not 100% sure though pls wait for profs reply

**Comment by lingyanhao** [comment]
@daviddl9 I don't agree with your answer. Your sequence diagram suggests that the return statement occurs outside the if-else block.

So the code corresponding it would look something like
`Task generate(boolean hasExisting) {`
`    Task t;`
`    if (hasExisting) {`
`        t = getTask();`
 `   } else {`
`        t = new Task();`
`    }`
`    return t;`
`}`
(Sorry for poor indentation, I don't know how to use proper indentation in markdown)

**Comment by fterh** [comment]
I agree with @lingyanhao and I feel @limli 's diagram is more representative of the original algorithm...

**Comment by ccristina** [comment]
Hi,

Indeed @limli's diagram is representative, while @daviddl9's diagram is incorrect when representing the return statements. 

@limli's diagram: the return arrows need to be longer, and end outside the `alt` box. 

Best regards,
Cristina

**Comment by daviddl9** [comment]
Okie thanks for the correction!
